### ISSUE TITLE
Error when updating pure: " Not a valid ref: refs/remotes/origin/main"

### ISSUE THREAD

**Issue by joestepp** [zimfw/zimfw#424]
Please check the existing issues to make sure you're not duplicating a report.

For bug reports, please provide the following information:

In a terminal, run `zimfw info` and paste the output below:

```
zimfw version: 1.4.1 (built at 2021-02-17 23:25:22 UTC, previous commit is 12dab4e)
ZIM_HOME:      /Users/joestepp/.zim
Zsh version:   5.8
System info:   Darwin Joe-Stepp-MBP16tb.local 19.2.0 Darwin Kernel Version 19.2.0: Sat Nov 9 03:47:04 PST 2019; root:xnu-6153.61.1~20/RELEASE_X86_64 x86_64
```

```
#Paste the output here
```

- gist of .zimrc: https://gist.github.com/joestepp/d34a6d0550410b8c313a5a3204bc2697
- gist of .zshrc: https://gist.github.com/joestepp/66c17e6061495a2d2df365d541cb0780

Description
-----------

I get the following error when running `zimfw update`:
```
❯ zimfw update   
x pure: Error during git remote set-head
  error: Not a valid ref: refs/remotes/origin/main
```
Steps to reproduce
------------------

Run ``zimfw update`.

Images or other information
---------------------------

**Comment by ericbn** [comment]
Hi @ joestepp. The "git remote set-head" step was just added in zimfw version 1.4.1. I cannot reproduce your issue. Here's how to manually try to reproduce it (repeating the "git remote set-head" step that `zimfw update` does):

```
% cd ${ZIM_HOME}/modules/pure
% command git remote set-head origin -a
origin/HEAD set to main
```

**Comment by ericbn** [comment]
@joestepp, looks like this is caused by master being renamed to main. Please check https://stackoverflow.com/a/66253232/2654518

**Comment by ericbn** [comment]
Just released version 1.4.2 with the fix for this.

`zimfw update` will handle the branch renaming automatically, without users having to manually fix this in the repositories.

@joestepp, thanks for reporting this!

**Comment by joestepp** [comment]
@ericbn Awesome, thanks for fixing this so quickly!
